docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md

```diff
@@ -9,25 +9,247 @@ menu:
+
+### Expected behavior
+
+The table describes what the expected behavior is when a read happens from a follower.
+
+| Conditions | Expected behavior |
+| :--------- | :---------------- |
+| yb_read_from_followers is true AND transaction is marked read-only | Read happens from the closest follower |
+| yb_read_from_followers is false OR transaction/statement is not read-only | Read happens from the leader |
+
+### Read from follower conditions
+
+- If the follower’s safe-time is at least `<current_time> - <staleness>`, then the follower may serve the read without any delay.
+
+- If the follower is not yet caught up to `<current_time> - <staleness>`, then the read will be redirected to a different replica transparently from the end-user. The end user may see a slight increase in latency depending on the location of the replica which satisfies the read.
+
+### Read-only transaction conditions
+
+Regarding marking the transaction as read only, a user can do one of the following:
+
+- `SET TRANSACTION READ ONLY` applies only to the current transaction block.
+- `SET SESSION CHARACTERISTICS AS TRANSACTION READ ONLY` applies the read-only setting to all statements and transaction blocks that follow.
+- `SET default_transaction_read_only = TRUE` applies the read-only setting to all statements and transaction blocks that follow.
+- `/*+ Set(transaction_read_only true) */ SELECT` applies only to the current transaction/select statement.
+
+## Examples
+
+This example uses follower reads since the **transaction** is marked read-only.
+
+```sql
+set yb_read_from_followers = true;
+start transaction read only;
+SELECT * from t WHERE k='k1';
+commit;
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+```
+
+This example uses follower reads since the **session** is marked read only.
+
+```sql
+set session characteristics as transaction read only;
+set yb_read_from_followers = true;
+SELECT * from t WHERE k='k1';
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+The following examples use follower reads since **pg_hint** can be used during PREPARE, CREATE FUNCTION, and SELECT to do follower reads.
+
+{{< note title="Note" >}}
+The pg_hint needs to be applied at the prepare/function-definition stage and not at the `execute` stage.
+{{< /note >}}
+
+```sql
+set session characteristics as transaction read write;
+set yb_read_from_followers = true;
+
+PREPARE select_stmt(text) AS
+/*+ Set(transaction_read_only on) */
+SELECT * from t WHERE k=$1;
+
+EXECUTE select_stmt(‘k1’);
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+```sql
+set session characteristics as transaction read write;
+set yb_read_from_followers = true;
+
+CREATE FUNCTION func() RETURNS text AS
+$$ /*+ Set(transaction_read_only on) */
+SELECT * from t WHERE k=1 $$ LANGUAGE SQL;
+CREATE FUNCTION
+SELECT func();
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+```sql
+set session characteristics as transaction read write;
```

Comment:
Why is this line needed?

And overall the simple SELECT with the embedded hint should be the first example for pg_hint_plan, instead of the more uncommon cases of PREPARE and CREATE FUNCTION. The flow needs to be simple to advanced rather than reverse.

Reply:
It is just to highlight that pg_hint can be used in a read_write session to override the read only property.

We don't need this line per se. The example would work even without this line. So we can remove this, and otehr places in line 125 and 107.

Also, agree with moving this example above line 107. We can show this example, then the PREPARE example and then the CREATE FUNCTION example.

Reply:
Thank you ! I'll make the change accordingly.